lms/djangoapps/courseware/field_overrides.py

```diff
@@ -171,83 +255,84 @@ def default(self, block, name):
         return self.fallback.default(block, name)
 
 
-class _OverridesDisabled(threading.local):
-    """
-    A thread local used to manage state of overrides being disabled or not.
-    """
-    disabled = ()
+class OverrideModulestoreFieldData(OverrideFieldData):
+    """Apply field data overrides at the modulestore level. No student context required."""
 
+    @classmethod
+    def wrap(cls, block, field_data):  # pylint: disable=arguments-differ
+        if cls.provider_classes is None:
+            cls.provider_classes = [
+                resolve_dotted(name) for name in settings.MODULESTORE_FIELD_OVERRIDE_PROVIDERS
+            ]
 
-_OVERRIDES_DISABLED = _OverridesDisabled()
+        if block:
+            enabled_providers = cls._providers_for_block(block)
+            if enabled_providers:
+                return cls(field_data, enabled_providers)
 
+        return field_data
 
-@contextmanager
-def disable_overrides():
-    """
-    A context manager which disables field overrides inside the context of a
-    `with` statement, allowing code to get at the `original` value of a field.
-    """
-    prev = _OVERRIDES_DISABLED.disabled
-    _OVERRIDES_DISABLED.disabled += (True,)
-    yield
-    _OVERRIDES_DISABLED.disabled = prev
+    @classmethod
+    def _providers_for_block(cls, block):
```

Comment:
Can you add some inline documentation/comments on what this function do? Any in particular, why it leverages `request_cache`?